proto/src/determined/api/v1/api.proto

```diff
@@ -580,6 +580,21 @@ service Determined {
       tags: "Internal"
     };
   }
+  // Report the receipt of a signal to stop the given allocation early.
+  // This is used to communicate back from a SLURM job that it has been
+  // notified of a pending preememption. Upon a call to this API
+  // the RM should then trigger a checkpoint and immediate exit.
+  rpc AllocationPendingPreemptionSignal(
+      AllocationPendingPreemptionSignalRequest)
+      returns (AllocationPendingPreemptionSignalResponse) {
+    option (google.api.http) = {
+      post: "/api/v1/allocations/{allocation_id}/signals/pending_preemption"
```

Comment:
maybe "ForwardPreemption"

Reply:
It is the forward of a pending preemption that will not happen for GraceTime seconds.   So not exactly a preemption now.  I was trying to capture that.